tesseract_motion_planners/ompl/src/ompl_motion_planner.cpp

```diff
@@ -104,439 +95,272 @@ bool OMPLMotionPlanner::terminate()
-        // The interpolate function only executes if the current number of states is less than the requested
-        p->simple_setup->simplifySolution();
-        if (p->simple_setup->getSolutionPath().getStateCount() < num_output_states)
-          p->simple_setup->getSolutionPath().interpolate(num_output_states);
-      }
+      // Trajectory length matches or exceeds seed length and simplification is disabled, so no need to simplify or
+      // interpolate
     }
   }
 
-  // Flatten the results to make them easier to process
-  response.results = request.seed;
-  std::vector<std::reference_wrapper<Instruction>> results_flattened =
-      flattenProgramToPattern(response.results, request.instructions);
-  std::vector<std::reference_wrapper<const Instruction>> instructions_flattened = flattenProgram(request.instructions);
-
-  std::size_t instructions_idx = 0;  // Index for each input instruction
+  // Get the results
+  tesseract_common::TrajArray trajectory = toTrajArray(simple_setup->getSolutionPath(), extractor);
+  assert(checkStartState(simple_setup->getProblemDefinition(), trajectory.row(0), extractor));
+  assert(checkGoalState(simple_setup->getProblemDefinition(), trajectory.bottomRows(1).transpose(), extractor));
 
-  // Handle the start instruction
-  const auto& plan_instruction = instructions_flattened.at(0).get().as<PlanInstruction>();
-  if (plan_instruction.isStart())
+  // Enforce limits
   {
-    const auto& p = problem[0];
+    const std::string manipulator = request.instructions.getManipulatorInfo().manipulator;
+    auto joint_limits = request.env->getJointGroup(manipulator)->getLimits().joint_limits;
+    for (Eigen::Index i = 0; i < trajectory.rows(); i++)
+      tesseract_common::enforcePositionLimits(trajectory.row(i), joint_limits);
+  }
 
-    // Get the results
-    tesseract_common::TrajArray trajectory = p->getTrajectory();
+  // Initialize the output by copying the input program and flattening it
+  response.results = request.seed;
+  std::vector<std::reference_wrapper<Instruction>> results_flattened =
+      flattenProgramToPattern(response.results, request.instructions);
 
-    // Enforce limits
-    for (Eigen::Index i = 0; i < trajectory.rows(); i++)
-      tesseract_common::enforcePositionLimits(trajectory.row(i), p->manip->getLimits().joint_limits);
+  // Element 0 of the flattened results is a start instruction that should contain the first trajectory state
+  auto& start_instruction = results_flattened.at(0).get().as<MoveInstruction>();
 
-    assert(checkStartState(p->simple_setup->getProblemDefinition(), trajectory.row(0), p->extractor));
-    assert(checkGoalState(p->simple_setup->getProblemDefinition(), trajectory.bottomRows(1).transpose(), p->extractor));
+  // Element 1 of the flattened results is a composite instruction containing the reset of the trajectory
+  auto& composite = results_flattened.at(1).get().as<CompositeInstruction>();
```

Comment:
It seems confusing to have a separate move instruction for the start of the trajectory and to have the rest of the trajectory inside a composite instruction, which itself has a start instruction placeholder which happens to be a `NullInstruction`. Is there a reason it was set up like this in the previous version? Would it be better to put the first trajectory waypoint into the start instruction of the composite with the rest of the trajectory?